src/app/pages/FastBtcPage/components/Withdraw/AmountForm.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useCallback, useContext, useMemo, useState } from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { bignumber } from 'mathjs';
+import { AssetSymbolRenderer } from '../../../../components/AssetSymbolRenderer';
+import { Asset } from '../../../../../types';
+import { translations } from 'locales/i18n';
+import { WithdrawContext, WithdrawStep } from '../../contexts/withdraw-context';
+import { useBalance } from '../../../../hooks/useBalance';
+import { AmountInput } from '../../../../components/Form/AmountInput';
+import { FormGroup } from 'app/components/Form/FormGroup';
+import { gasLimit } from '../../../../../utils/classifiers';
+import { TxType } from '../../../../../store/global/transactions-store/types';
+import { WithdrawDetails } from './WithdrawDetails';
+import { FastBtcButton } from '../FastBtcButton';
+import { weiToNumberFormat } from '../../../../../utils/display-text/format';
+import { LoadableValue } from '../../../../components/LoadableValue';
+import { NetworkAwareComponentProps } from '../../types';
+import { getBTCAssetForNetwork } from '../../helpers';
+import { btcInSatoshis } from 'app/constants';
+
+export const AmountForm: React.FC<NetworkAwareComponentProps> = ({
+  network,
+}) => {
+  const { amount, limits, set } = useContext(WithdrawContext);
+  const { t } = useTranslation();
+
+  const balance = useBalance();
+
+  const [value, setValue] = useState(amount);
+
+  const invalid = useMemo(() => {
+    const amount = value || '0';
+    const weiAmount = Number(amount) * btcInSatoshis;
```

Comment:
This is not correct:
1. `amount` is already `Number` so no need to cast it
2. it's not wei amount but amount in satoshis

Since both `limits.min` and `limits.max` seem to be in satoshis as well, probably just rename it to `satoshiAmount` and don't cast `amount` as `Number`.

Cleaner solution would be to have limits in `wei` and use `toWei(amount)` here.